src/app/components/card-list/card-list.component.ts

```diff
@@ -0,0 +1,154 @@
+        map((pokemons) => pokemons)
+      );
+    }
+  }
+
+  filterData(data: SearchBarEventArgs) {
+    this.queryParams = data.newValue;
+    this.reload();
+  }
+
+  addToFavorites(pokemon: PokemonsPage) {
+    let favoritesLength = undefined;
+    let removingFavorite = false;
+
+    this.favoriteEntityService.entities$.subscribe((entities) => {
+      favoritesLength = entities.length;
+
+      if (entities.find((item) => item.name == pokemon.name)) {
+        removingFavorite = true;
+      }
+    });
+    if (removingFavorite) {
+      const newPokemon = { ...pokemon, isFavorite: false };
+      this.pokemonsEntityService.updateOneInCache(newPokemon);
+      this.favoriteEntityService.removeOneFromCache(pokemon);
+    } else {
+      if (favoritesLength !== undefined && favoritesLength >= 5) {
+        console.log(`ya no agregamos mas`);
+      } else {
+        const newPokemon = { ...pokemon, isFavorite: true };
+        this.pokemonsEntityService.updateOneInCache(newPokemon);
+        this.favoriteEntityService.addOneToCache(pokemon);
+      }
+    }
+  }
+
+  loadMorePokemons() {
+    this.pokemonsEntityService.getWithQuery({
+      limit: '20',
+      id: this.pokemonApiOffset,
+    });
+    let nextOffsetInt = parseInt(this.pokemonApiOffset) + 20;
+    this.pokemonApiOffset = nextOffsetInt + '';
+  }
+
+  async openPokemonDialog(pokemon) {
+    const dialogConfig = defaultDialogConfig();
+
+    let pokemonData: Pokemon = undefined;
+    this.fetchedPokemonsEntityService.entities$.subscribe((entity) => {
+      let data = entity.filter((item) => item.name === pokemon.name);
+      pokemonData = data[0];
+    });
+
+    if (
+      pokemonData === undefined ||
+      pokemonData.name == undefined ||
+      !pokemonData
+    ) {
+      this.loader.start();
+      pokemonData = await this.fetchedPokemonsEntityService
+        .getByKey(pokemon.url)
+        .toPromise();
+      this.loader.stop();
+    }
+
+    if (this.isComparing) {
+      dialogConfig.data = {
+        pokemon1: this.pokemonBeforeComparing,
+        pokemon2: { ...pokemonData, photo: pokemon.photo },
+      };
+      this.dialog
+        .open(MultipleCardOverviewComponent, dialogConfig)
+        .afterClosed()
+        .subscribe((data: Pokemon) => {
+          if (data !== null) {
+            this.pokemonBeforeComparing = data;
+            this.isComparing = true;
+          } else {
+            this.isComparing = false;
+          }
+        });
+    } else {
+      dialogConfig.data = {
+        pokemonPageInfo: pokemon,
+        pokemon: pokemonData,
+      };
+
+      this.dialog
+        .open(SingleCardOverviewComponent, dialogConfig)
+        .afterClosed()
+        .subscribe((data: Pokemon) => {
+          if (data !== null) {
+            this.pokemonBeforeComparing = data;
+            this.isComparing = true;
+          } else {
+            this.isComparing = false;
+          }
+        });
+    }
```

Comment:
I recommend splitting this long function into multiple small functions that explain what they are doing with their name